datafusion-examples/examples/expr_api.rs

```diff
@@ -518,10 +518,7 @@ fn type_coercion_demo() -> Result<()> {
         HashMap::new(),
     )?;
     let i8_array = Int8Array::from_iter_values(vec![0, 1, 2]);
-    let batch = RecordBatch::try_new(
-        Arc::new(df_schema.as_arrow().to_owned()),
-        vec![Arc::new(i8_array) as _],
-    )?;
+    let batch = RecordBatch::try_new((&df_schema).into(), vec![Arc::new(i8_array) as _])?;
```

Comment:
I would personally find it easier to understand there were no `clone`s going on using this syntax (I don't think this is actually any different code):

```suggestion
    let batch = RecordBatch::try_new(Arc::clone(&df_schema.as_ref()), vec![Arc::new(i8_array) as _])?;
```